### ISSUE TITLE
Missing Keras import guard in KerasClassifier's custom_loss_gradient function.

### ISSUE THREAD

**Issue by hhuangMITRE** [Trusted-AI/adversarial-robustness-toolbox#809]
Hello! While we were testing one of the ART poisoning attacks (Feature Collision) we encountered an import issue with the `self.estimator.custom_loss_gradient` function:

Line 633 of `keras.py` (master branch):
 `        import keras.backend as k`

We're currently using TensorFlow v2 which uses `import tf.keras` commands instead. 
We fixed the issue locally by swapping line 633 with the following lines :

```
        if self.is_tensorflow:
            import tensorflow.keras.backend as k
        else:
            import keras.backend as k
```

Let me know if this correction can be applied into `keras.py`, thanks! 
We're currently using ART release 1.5. 
Let me know if this change can be applied into the 1.5.x version, or if it will apply in release 1.6.

Thanks,
Howard

**Comment by ebubae** [comment]
Thanks for pointing out this issue, and posting your solution. We'll try to get this resolved for the 1.5.1 release or later. Track progress with this PR: #810 

Thanks again for using ART!

**Comment by hhuangMITRE** [comment]
You're welcome. Happy to help!
